src/selectors/course.js

```diff
@@ -8,33 +8,40 @@ const getIconContext = (state) => state.context.iconContext;
 // Creates a list of courses with lessons that have tags matching the filter
 export const getFilteredCourses = createSelector(
   [getFilteredLessons, getPlaylists, getIconContext],
-  (lessons = {}, playlists = {}, iconContext) => {
+  getCourses
+);
+
+// Creates a list of courses with indexed lessons that have tags matching the filter
+export const getFilteredAndIndexedCourses = createSelector(
+  [getFilteredAndIndexedLessons, getPlaylists, getIconContext],
+  getCourses
+);
 
-    return Object.keys(lessons).reduce((res, lessonKey) => {
-      const lesson = lessons[lessonKey];
-      const courseName = lesson.course;
-      const name = capitalize(courseName).replace('_', ' ');
+// Create course objects
+export function getCourses(lessons = {}, playlists = {}, iconContext){
```

Comment:
Egentlig trenger ikke getCourses å exporteres, ettersom det er bare en hjelpefunksjon for de to selectorene over, men jeg ser at du gjør det likevel, regner med det er sånn at du kan teste getCourses direkte, istedetfor å måtte velge én av selektorene over og teste den (eller begge, som ville vært litt meningsløst). Så jeg forstår hvorfor du gjorde det slik. Men det føles ikke helt riktig. Så jeg begynte å tenke litt.

Selv om du gjorde akkurat det jeg foreslo, så virker oppsettet nå litt ... off. F.eks. så får PlaylistPage inn alle kursene, og så bruker den params.course for å velge ut ett av dem. Hva om course-selectorene fikk inn params.course som en prop, og sendte bare tilbake det riktige kurset? (https://github.com/reactjs/reselect#accessing-react-props-in-selectors) 

Den vil riktignok rekalkulere hver gang får inn en ny prop (altså hver gang man går til playlistpage med nytt kurs), men det er vel ikke så farlig. Ellers må man følge dette: https://github.com/reactjs/reselect#sharing-selectors-with-props-across-multiple-components

Men dette løser ikke det jeg nevnte først. 

En interessant observasjon fra PlaylistPage er at det bare er playlists som brukes av filteredCourse.
Og fra filteredAndIndexedCourses brukes bare lessons.
Hadde det ikke vært bedre om PlaylistPage bare brukte selectorene getPlaylists og getFilteredLessons direkte, der de får inn courseName som prop?
I såfall er det bare FrontPage som får getFilteredCourses, og ingen bruker getFilteredAndIndexedCourses. Og da kan man redusere antall selectorer i courses.js til 1.

Det eneste som føles litt dumt ut da, er at courses bare finnes i én versjon, nemlig den der lessons bare er filtrert, men indexed:false ikke er luket ut. Men er det riktig/lurt at lessons henger i courses i det hele tatt? Det er jo da bare CourseItem som bruker disse lessons, og det eneste som brukes er lessons.length. Og dette kunne man jo lagret i courses explisitt.

Så da står man igjen med courses som ikke inneholder lessons (men antall filtered lessons), en selector getFilteredLessons som brukes av PlaylistNavigation, getFilteredAndIndexLessons som brukes av PlaylistPage.

Hva tror du? Gir det mening? Skriver jeg forståelig?


Forøvrig litt pussig dette med indexed lessons, for på forsiden så bruker man bare filtered lessons når man viser antall, og så hvis da mange av disse er indexed:false, så vil de ikke dukke opp i midten, bare i playlists... burde kanskje tenke gjennom hva som er meningen med playlists, eller altså, om det er en bedre måte å få frem det behovet de er ment å dekke? Dette siste trenger vi ikke ta her, men burde kanskje bli opprettet en egen issue på det? Eller lager jeg trøbbel av noe som er uproblematisk?